src/EventGridEmulator/Configuration/TopicOptions.cs

```diff
@@ -19,5 +21,60 @@ public TopicOptions(TopicOptions original)
         }
     }
 
-    public Dictionary<string, string[]> Topics { get; set; } = new Dictionary<string, string[]>();
+    public Dictionary<string, string[]> Topics { get; set; } = new Dictionary<string, string[]>(StringComparer.OrdinalIgnoreCase);
+
+    public override bool Equals(object? obj)
+    {
+        if (ReferenceEquals(this, obj))
+        {
+            return true;
+        }
+
+        return obj is TopicOptions other && this.Equals(other);
+    }
+
+    private bool Equals(TopicOptions other) => (this.Topics, other.Topics) switch
+    {
+        (null, null) => true,
+        (not null, null) => false,
+        (null, not null) => false,
+        (not null, not null) => this.TopicsEquals(other),
+    };
+
+    private bool TopicsEquals(TopicOptions other)
+    {
+        if (this.Topics.Count != other.Topics.Count)
+        {
+            return false;
+        }
+
+        foreach (var (topicName, subscribers) in this.Topics)
+        {
+            if (!other.Topics.TryGetValue(topicName, out var otherSubscribers))
+            {
+                return false;
+            }
+
+            if (!subscribers.SequenceEqual(otherSubscribers, StringComparer.Ordinal))
+            {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    [SuppressMessage("ReSharper", "NonReadonlyMemberInGetHashCode", Justification = "This is a DTO, also we don't plan on storing this in a hash table.")]
+    public override int GetHashCode()
+    {
+        var hash = 17;
+
+        foreach (var pair in this.Topics)
+        {
+            hash = hash * 23 + pair.Key.GetHashCode();
```

Comment:
I'm guessing 17 * 23 and all is usual hashcode stuff?

Reply:
Yep usual prime number / hashcode stuff, but thx to your comment I remembered to add an `unckecked` block